### PR TITLE
WIP: Add header and footer api

### PR DESCRIPTION
**Pull request by tomdoughty**
## Description

Build header and footer from NHSUK header and footer API

## Related Issue

https://jira.service.nhs.uk/browse/TM-1240

## Motivation and Context

Adds consistency to NHS.UK applications

## Checklist

- [x] An [ADR](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
- [ ] New and/or updated tests
- [x] Changes log in `CHANGELOG`

### CODE REVIEW

File: app/middleware/headerAndFooter.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      res.locals = {
+        ...res.locals,
+        footerLinks: cachedData.footerLinks,
+        headerLinks: cachedData.headerLinks,
+      };
+      return next();
+    }
+
+    // Get data from Wagtail endpoint
+    const data = await rp('https://www.nhs.uk/content-api/header-footer.json');
```
Comment (anlinhs) [inline app/middleware/headerAndFooter.js:21]: Can the API URL be a variable so we can use integration, staging and live?

Comment (tomdoughty) [reply app/middleware/headerAndFooter.js:21]: This is still a WIP atm as we need to discuss with people how to get around the basic auth on lower environments

Comment (anlinhs) [reply app/middleware/headerAndFooter.js:21]: Mike pointed me to these URLs which are not behind basic auth:
https://nhsuk-live.nhswebsite-integration.nhs.uk/content-api/header-footer.json
https://nhsuk-live.nhswebsite-staging.nhs.uk/content-api/header-footer.json

Comment (tomdoughty) [reply app/middleware/headerAndFooter.js]: Default of the integration URL has been added so it is easier for local development... I will add staging to Vault, infra task can be added as dependency of release for production env var to be added

**Pull request closed without merging**